packages/airnode-protocol/contracts/rrp/authorizers/admin/AirnodeTokenPayment.sol

```diff
@@ -0,0 +1,282 @@
+            "Invalid destination address"
+        );
+        airnodeToPaymentDestination[msg.sender] = _paymentDestination;
+        emit SetAirnodeToPaymentDestination(_paymentDestination, msg.sender);
+    }
+
+    /// @notice Make payments to gain access to Airnode
+    /// @dev In order for this function to be able to extend the whitelisting,
+    /// the Airnode operator must first grant the whitelist expiration extender
+    /// role to this contract, otherwise it will revert.
+    /// chainId-airnode-endpoint-requester pair gets authorized for the
+    /// duration specified. The amount to be paid is determined by the fee set
+    /// in the AirnodeFeeRegistry contract and the token price.
+    /// @param _chainId Id of the chain
+    /// @param _airnode Airnode address
+    /// @param _endpointId Id of the endpoint
+    /// @param _requesterAddress Requester address
+    /// @param _whitelistDuration Duration in seconds for which the requester
+    /// will be whitelisted
+    function makePayment(
+        uint256 _chainId,
+        address _airnode,
+        bytes32 _endpointId,
+        address _requesterAddress,
+        uint64 _whitelistDuration
+    ) external override {
+        require(_chainId != 0, "Invalid chainId");
+        require(_airnode != address(0), "Invalid Airnode address");
+        require(_requesterAddress != address(0), "Invalid requester address");
+        require(
+            _whitelistDuration != 0 &&
+                _whitelistDuration <= DEFAULT_MAXIMUM_WHITELIST_DURATION &&
+                _whitelistDuration >= getMinimumWhitelistDuration(_airnode),
+            "Invalid whitelist duration"
+        );
+
+        address requesterAuthorizerWithManager = IAirnodeRequesterAuthorizerRegistry(
+                airnodeRequesterAuthorizerRegistry
+            ).chainIdToRequesterAuthorizerWithManager(_chainId);
+        require(
+            requesterAuthorizerWithManager != address(0),
+            "No RequesterAuthorizersWithManager set for chainId"
+        );
+
+        (
+            uint64 expirationTimestamp,
+            uint192 indefiniteWhitelistCount
+        ) = IRequesterAuthorizerWithManager(requesterAuthorizerWithManager)
+                .airnodeToEndpointIdToRequesterToWhitelistStatus(
+                    _airnode,
+                    _endpointId,
+                    _requesterAddress
+                );
+        require(
+            indefiniteWhitelistCount == 0,
+            "Already whitelisted indefinitely"
+        );
+        require(
+            uint64(block.timestamp) + _whitelistDuration > expirationTimestamp,
+            "Already whitelisted"
+        );
+
+        uint256 amount = getPaymentAmount(
+            _chainId,
+            _airnode,
+            _endpointId,
+            expirationTimestamp > uint64(block.timestamp)
+                ? (uint64(block.timestamp) + _whitelistDuration) -
+                    expirationTimestamp
+                : _whitelistDuration
```

Comment:
We should only charge for the delta between the existing expiration and current whitelist duration.

Reply:
I'm now thinking that this should go inside the `getPaymentAmount()` function 🤔

Reply:
if we go by what I mentioned above we won't need to do this and just have to calculate for `_whitelistDuration`